javascript/012-lasagna-master/lasagna-master.js

```diff
@@ -0,0 +1,76 @@
+/// <reference path="./global.d.ts" />
+// @ts-check
+
+/**
+ * Determine whether the lasagna is done
+ * @param {number} timer
+ * @return {string} lasagna status
+ *
+ */
+
+export function cookingStatus(timer) {
+  if (timer === undefined) {
+    return 'You forgot to set the timer.';
+  }
+
+  return timer > 0 ? 'Not done, please wait.' : 'Lasagna is done.';
+}
+
+/**
+ * Estimate lasagna preparation time
+ * @param {string[]} layers of ingredients
+ * @param {number} timer
+ * @return {number} average preparation time
+ *
+ */
+export function preparationTime(layers, timer = 2) {
+  return layers.length * timer;
+}
+
+/**
+ * Compute the amounts of noodles and sauce needed
+ * @param {string[]} ingredients
+ * @return {object} quantity of ingredients
+ *
+ */
+export function quantities(ingredients) {
+  const noodlesLayerQuantity = 50;
+  const sauceLayerQuantity = 0.2;
+
+  return {
+    noodles: ingredients.filter(x => x === 'noodles').length * noodlesLayerQuantity,
+    sauce: ingredients.filter(x => x === 'sauce').length * sauceLayerQuantity,
```

Comment:
Nit: This is grossly inefficient because you're creating two new arrays just to calculate their length and discard them.
This, however, is good enough for this task, albeit it's a shame JS doesn't have a `count()` function